virtcontainers/clh.go

```diff
@@ -437,7 +434,13 @@ func (clh *cloudHypervisor) hotplugAddBlockDevice(drive *config.BlockDrive) erro
 			VhostUser: false,
 			Id:        driveID,
 		}
-		_, _, err = cl.VmAddDiskPut(ctx, blkDevice)
+		var deviceInfo chclient.PciDeviceInfo
+		deviceInfo, _, err = cl.VmAddDiskPut(ctx, blkDevice)
+
+		// Set the PCIAddr as the PCI BDF returned from clh
+		drive.PCIAddr = deviceInfo.Bdf
```

Comment:
I don't think this is quite right.  drive.PCIAddr is not actually a BDF address - which is confusing, because that's usually what "PCI address" means.  However at least for qemu, drive.PCIAddr is actually "<bridge slot>/<device slot>", and I'm assuming that's how the agent expects it.

I'm also a bit confused as to how cloud-hypervisor can tell us the guest Bdf from the host side: the "B" part depends on the guest OS's enumeration of the bus, so we can't generally predict that reliably from outside.

Reply:
@dgibson I agree with you that the name of `PCIAddr` is confusing. The counter part in kata-agent, `drive.Id`, makes more sense to me. This patch indeed wants to generalize the way how kata-agent can locate drives in the guest. Instead of relying solely on the format that is tied with QEMU ("bridge slot/device slot"), I feel BDF is a more generic format to locate devices in general. Of course, it requires changes in the agent side (to accept `drive.Id` in BDF format) and the PR is here https://github.com/kata-containers/agent/pull/824. I hope it clarifies the confusions.

That's a good question on how the BDF information is retrieved reliably by cloud-hypervisor.  I would ask @sboeuf to comment on it.

Reply:
>I'm also a bit confused as to how cloud-hypervisor can tell us the guest Bdf from the host side: the "B" part depends on the guest OS's enumeration of the bus, so we can't generally predict that reliably from outside.

Well CH implements only one PCI bus for now, so B is pretty easy as it's always 0. But globally you can control this kind of thing through ACPI.

Reply:
@likebreath sorry, I missed your link to the agent PR above.  That makes much more sense now.

> Well CH implements only one PCI bus for now, so B is pretty easy as it's always 0. 

So no PCI to PCI bridges, and no PCI-E supported?  The latter seems pretty restrictive.

> But globally you can control this kind of thing through ACPI.

I don't really see quite how that can work.